pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,863 @@
+                state["precip_decomp"][j]["cascade_levels"]
+            )
+
+        precip_forecast = params["recomp_method"](state["precip_decomp"][j])
+
+        if params["domain"] == "spectral":
+            precip_forecast = state["fft_objs"][j].irfft2(precip_forecast)
+
+        # Apply the precipitation mask
+        if params["mask_method"] is not None:
+            precip_forecast = self._apply_precipitation_mask(
+                precip_forecast, j, state, params
+            )
+
+        # Adjust the CDF of the forecast to match the observed precipitation field
+        if params["probmatching_method"] == "cdf":
+            precip_forecast = probmatching.nonparam_match_empirical_cdf(
+                precip_forecast, params["precip"]
+            )
+        # Adjust the mean of the forecast to match the observed mean
+        elif params["probmatching_method"] == "mean":
+            mask = precip_forecast >= params["precip_thr"]
+            mu_fct = np.mean(precip_forecast[mask])
+            precip_forecast[mask] = precip_forecast[mask] - mu_fct + params["mu_0"]
+
+        # Update the mask for incremental method
+        if params["mask_method"] == "incremental":
+            state["mask_prec"][j] = nowcast_utils.compute_dilated_mask(
+                precip_forecast >= params["precip_thr"],
+                params["struct"],
+                params["mask_rim"],
+            )
+
+        # Apply the domain mask (set masked areas to NaN)
+        precip_forecast[params["domain_mask"]] = np.nan
+
+        return precip_forecast
+
+    def _apply_precipitation_mask(self, precip_forecast, j, state, params):
+        """
+        Apply the precipitation mask to prevent new precipitation from generating
+        in areas where it was not observed.
+        """
+        precip_forecast_min = precip_forecast.min()
+
+        if params["mask_method"] == "incremental":
+            precip_forecast = (
+                precip_forecast_min
+                + (precip_forecast - precip_forecast_min) * state["mask_prec"][j]
+            )
+            mask_prec_ = precip_forecast > precip_forecast_min
+        else:
+            mask_prec_ = state["mask_prec"]
+
+        # Set to min value outside the mask
+        precip_forecast[~mask_prec_] = precip_forecast_min
+
+        return precip_forecast
+
+    def _measure_time(self, label, start_time):
+        """
+        Measure and print the time taken for a specific part of the process.
+
+        Parameters:
+        - label: A description of the part of the process being measured.
+        - start_time: The timestamp when the process started (from time.time()).
+        """
+        if self.measure_time:
+            elapsed_time = time.time() - start_time
+            print(f"{label} took {elapsed_time:.2f} seconds.")
+
+
+# Wrapper function to preserve backward compatibility
 @deprecate_args({"R": "precip", "V": "velocity", "R_thr": "precip_thr"}, "1.8.0")
```

Comment:
Looks like it's about time to deprecate these arguments. Would you do the honors?

Reply:
I will do this once I have also changed the blending/steps. This gives me some grasp of the old names to keep consistency. Once both are fully refactored, this will be the first thing on the list. 
I first wanted some feedback regarding the splitting of the steps function before I tackle the blending part because that code is quite a bit more complicated.